### PR TITLE
Show filtered total in default currency

### PR DESCRIPTION
**Pull request by Gataniel**

### CODE REVIEW

File: app/models/transaction.rb [hunk 1]
```diff
@@ -51,15 +51,34 @@ class Transaction < ActiveRecord::Base
...
-    ]
+  class << self
+    def flow_ordered(def_currency)
+      currencies = Currency.ordered(def_currency)
+
+      curr_flow = {}
+      currencies.each do |curr|
+        curr_flow[curr] = {}
+
+        transactions = by_currency(curr)
```
Comment (shir) [inline app/models/transaction.rb]: @Gataniel I think it will be better to make this via sql with one query. It will work much faster.

Comment (shir) [inline app/models/transaction.rb]: @Gataniel and create separate object/structure to store flow.

File: app/views/home/_flow.html.slim [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
+    .row
       | Total:
-    .col-sm-2 class=(@rub_flow > 0 ? 'bg-success' : 'bg-danger')
-      = money_with_symbol(@rub_flow)
-    .col-sm-2 class=(@usd_flow > 0 ? 'bg-success' : 'bg-danger')
-      = money_with_symbol(@usd_flow)
+  - def_curr = current_organization.default_currency
+  - @curr_flow.each_pair do |curr, calc|
+    .col-sm-2
+      .row class=(calc[:inc] > 0 ? 'bg-success' : 'bg-danger')
```
Comment (shir) [inline app/views/home/_flow.html.slim]: @Gataniel put class calculation in helper.

File: spec/models/transaction_spec.rb [hunk 3]
```diff
@@ -214,4 +214,29 @@
...
+    let(:def_trans)   { org.transactions.by_currency(def_curr) }
+
+    let(:slave_inc) { Money.new(slave_trans.incomes.sum(:amount_cents), slave_curr)}
+    let(:slave_exp) { Money.new(slave_trans.expenses.sum(:amount_cents), slave_curr)}
+    let(:def_inc) { Money.new(def_trans.incomes.sum(:amount_cents), def_curr)}
+    let(:def_exp) { Money.new(def_trans.expenses.sum(:amount_cents), def_curr)}
+
+    subject { org.transactions.flow_ordered(org.default_currency) }
+
+    it "contain right values" do
```
Comment (shir) [inline spec/models/transaction_spec.rb]: @Gataniel it's bad description.

File: app/helpers/money_helper.rb [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
 
-  def money_sign(money)
-    money > 0 ? 'success' : 'danger'
-  end
-
   def money_with_symbol(money)
     humanized_money_with_symbol(money, symbol_after_without_space: true)
   end
+
+  def bg(amount)
```
Comment (AlexanderUlitin) [inline app/helpers/money_helper.rb]: @Gataniel please name this method more clear, like `amount_bg_class` or something similar.

File: app/views/home/_flow.html.slim [hunk 5]
```diff
@@ -1,24 +1,25 @@
 #flow.h4
-  .row
-    .col-sm-2
+  .col-sm-2
```
Comment (shir) [inline app/views/home/_flow.html.slim]: @Gataniel you should use table here. Use table to show table data

**Pull request merged**